tools/db_crashtest.py

```diff
@@ -909,7 +916,11 @@ def blackbox_crash_main(args, unknown_args):
     # Print stats of the final run
     print("stdout:", outs)
 
-    print_if_stderr_has_errors(errs)
+    # Print stderr of the final run
+    if args.print_stderr_separately:
+        print(errs, file=sys.stderr)
```

Comment:
Can we call print_if_stderr_has_errors and make decision there to print in a file or on std?

Reply:
I thought about doing so first, but then there will be some existing behavior changes. If we are okay with the existing behavior changes, I'm all up for it.

Here are some of the examples that require changes.

- In line 889 before the change, we simply print out `stderr` as is without calling `print_if_stderr_has_errors()` and `exit(2)`.
- We also print `stderr` as is without calling `print_if_stderr_has_errors()` in the whitbox test (line 1057 before the change)

If we want `print_if_stderr_has_errors()` to make the decision then we'd want to replace all `print(stderr)` with `print_if_stderr_has_errors()` first so that we don't miss anything.

Reply:
I think change is fine. All we want is to print stderr. So even though if it exit(2), we want to make sure it print errors. In print_if_stderr_has_errors function you can change the function to print directly instead of  doing 
```
for line in stderr.split("\n"):
        if line != "" and not line.startswith("WARNING"):
            print("stderr has error message:")
            print("***" + line + "***")
```